### PR TITLE
Match retail z_std_dma.c

### PR DESCRIPTION
**Pull request by cadmic**
Notably, calls to `DmaMgr_Error` were replaced with `Fault_AddHungupAndCrash`, which takes a file and line number.

### CODE REVIEW

File: src/boot/z_std_dma.c [hunk 1]
```diff
@@ -229,7 +235,7 @@ void DmaMgr_DmaFromDriveRom(void* ram, uintptr_t rom, size_t size) {
  *
  * This function does not return.
  */
-NORETURN void DmaMgr_Error(DmaRequest* req, const char* file, const char* errorName, const char* errorDesc) {
+NORETURN void DmaMgr_Error(DmaRequest* req, const char* filename, const char* errorName, const char* errorDesc) {
     uintptr_t vrom = req->vromAddr;
```
Comment (fig02) [inline src/boot/z_std_dma.c:239]: we use `file` way more than `filename`, at least currently. We can discuss fixing the inconsistency but it should prob be cleaned up in one pass
fwiw file is prob used more often for symmetry with the `__FILE__` directive

Comment (cadmic) [reply src/boot/z_std_dma.c:239]: ty

**Review by fig02 — APPROVED** [review]

File: src/boot/z_std_dma.c [hunk 2]
```diff
@@ -132,27 +135,29 @@ s32 DmaMgr_DmaRomToRam(uintptr_t rom, void* ram, size_t size) {
         ram = (u8*)ram + buffSize;
     }
 
-    if (1) {} // Also necessary to match
+    if (1) { // Also necessary to match
+        s32 pad[2];
```
Comment (Dragorn421) [inline src/boot/z_std_dma.c:140]:
```suggestion
    if (1) { // Also necessary to match
        s32 pad[2];
    }
```
would prefer to not indent the whole thing if we can avoid it

File: src/boot/z_std_dma.c [hunk 3]
```diff
@@ -460,11 +481,15 @@ s32 DmaMgr_RequestAsync(DmaRequest* req, void* ram, uintptr_t vrom, size_t size,
...
-    if ((1 && (ram == NULL)) || (osMemSize < OS_K0_TO_PHYSICAL(ram) + size) || (vrom & 1) || (vrom > 0x4000000) ||
+#if OOT_DEBUG
+    if ((ram == NULL) || (osMemSize < OS_K0_TO_PHYSICAL(ram) + size) || (vrom & 1) || (vrom > 0x4000000) ||
         (size == 0) || (size & 1)) {
-        //! @bug `req` is passed to `DmaMgr_Error` without rom, ram and size being set
-        DmaMgr_Error(req, NULL, "ILLIGAL DMA-FUNCTION CALL", "パラメータ異常です");
+        // The line numbers for `DMA_ERROR` are only used in retail builds, but this usage was removed so
+        // its line number is unknown.
+        //! @bug `req` is passed to `DMA_ERROR` without rom, ram and size being set
+        DMA_ERROR(req, NULL, "ILLIGAL DMA-FUNCTION CALL", "パラメータ異常です", "../z_std_dma.c", 0);
```
Comment (Dragorn421) [inline src/boot/z_std_dma.c:490]: why not just use `DmaMgr_Error` here

**Review by Dragorn421 — COMMENTED** [review]
I'm confused by `sDmaMgrIsRomCompressed` being set only in debug code, when it's also used in retail code (and it's also initialized to false in retail)

maybe it's a bug (doesn't really matter either way, towards the end of `DmaMgr_ProcessRequest` either the rom hangs up properly or most likely misbehaves and crashes)

File: src/boot/z_std_dma.c (hunk 2 above)
Comment (cadmic) [reply src/boot/z_std_dma.c]: I thought that the padding was some optimized-out local variables for the following code, and maybe we could figure out what they were later, but I don't mind the unindent

File: src/boot/z_std_dma.c (hunk 3 above)
Comment (cadmic) [reply src/boot/z_std_dma.c:490]: I feel that is less consistent with the rest of the code

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**